App/dfu/screens/FileSectionView.swift

```diff
@@ -90,28 +94,46 @@ struct FileSectionView: View {
     }
     
     private func onFileOpen(opened fileUrl: URL) {
-        do {
-            print(fileUrl)
+        let downloadTask = URLSession.shared.downloadTask(with: fileUrl) {
+            urlOrNil, responseOrNil, errorOrNil in
+            if let error = errorOrNil {
+                viewModel.onFileError(message: error.localizedDescription)
+                return
+            }
             
-            guard fileUrl.startAccessingSecurityScopedResource() else { return }
-            defer { fileUrl.stopAccessingSecurityScopedResource() }
+            if let response = responseOrNil as? HTTPURLResponse {
+                guard response.statusCode >= 200 && response.statusCode < 300 else {
+                    viewModel.onFileError(message: "Downloading file failed")
+                    return
+                }
+            }
+            guard let response = responseOrNil else {
+                viewModel.onFileError(message: "File nor found")
+                return
+            }
             
-            let fileHelper = FileHelper()
-            let fileRes = try fileUrl.resourceValues(forKeys:[.fileSizeKey, .nameKey])
-            let fileName = fileRes.name!
-            if let tmpFile = fileHelper.copyFileToTmpDirectory(fileName: fileName, readFromDisk: fileUrl) {
-                let resources = try tmpFile.resourceValues(forKeys:[.fileSizeKey, .nameKey])
+            guard let fileURL = urlOrNil else { return }
```

Comment:
I learned about this relatively recently, in case down the line you need something similar regarding temporary files: https://nshipster.com/temporary-files/